components-core/src/main/java/com/airwallex/android/core/Airwallex.kt

```diff
@@ -704,7 +704,45 @@ class Airwallex internal constructor(
             options,
             object : PaymentListener<PaymentIntent> {
                 override fun onSuccess(response: PaymentIntent) {
-                    listener.onCompleted(AirwallexPaymentStatus.Success(response.id))
+                    if (response.nextAction != null) {
+                        val provider = AirwallexPlugins.getProvider(ActionComponentProviderType.GOOGLEPAY)
+                        if (provider == null) {
+                            listener.onCompleted(
+                                AirwallexPaymentStatus.Failure(
+                                    AirwallexCheckoutException(message = "Missing ${PaymentMethodType.GOOGLEPAY.dependencyName} dependency")
+                                )
+                            )
+                            return
+                        }
+                        provider.get().retrieveSecurityToken(
```

Comment:
I'm just thinking whether we could simply put this inside GooglePayComponent

Reply:
Good point, and actually `retrieveSecurityToken` needs to be called before confirming intent as well, so I refactored to the Extension `ActionComponentProvider+Extensions.kt`